### ISSUE TITLE
Add scaladoc as description in the schema

### ISSUE THREAD

**Issue by bjornbak** [andyglow/scala-jsonschema#18]
I understand you want to keep the API as strictly by-convention so the same case classes can be used for multiple serializations.

IntelliJ adds popups with the json schema descriptions when work with json validated with a schema.

It would be a helpful and very useful feature if you took the Scaladoc for the class on the objects and @param for the individual fields.

**Comment by andyglow** [comment]
Hey, Christian! I am not quite sure I understand what kind of functionality are you looking for. Can you provide some more details possibly with examples?

**Comment by bjornbak** [comment]
I would like scaladoc'ed case class like
```
/**
 * This is my brilliant case class with multiple parameters
 * @param param1 This is the first param
 * @param param2 This is the second important param
 */
case class Brilliant(param1: Boolean, param2: String) extends Model
```

To result in description being added to the object and the properties:
```
{
    "type" : "object",
    "additionalProperties" : false,
    "description": "This is my brilliant case class with multiple parameters",
    "properties" : {
      "param1" : {
        "type" : "boolean",
        "description": "This is the first param"
      },
      "name" : {
        "type" : "string",
        "description": "This is the second important param"
      },
```

So that after the @param should be added as description to the property while everything else is added to the object itself.

**Comment by andyglow** [comment]
@bjornbak Hey, Christian! Some initial support for described feature has been added in version `0.2.7-M1`
Actually described here case is covered.

**Comment by bjornbak** [comment]
This is great news! I'll give it a try.

**Comment by bjornbak** [comment]
I can't make it work.. :(
I am not certain how to use it...

I found the changelog. 
What are the `add scaladoc support for sum types` you refer to?

here is some problem. comments live in source files and get erased after compilation, so resulted bytecode doesn't have this info and thus schemas derived from such models won't have descriptions. this might probably be solved by packing scaladocs into annotations by macro or compiler plugin and have those reused my SchemaMacro. Here is an example of something like this: https://medium.com/@takezoe/a-compiler-plugin-makes-scaladoc-readable-at-runtime-aecbebccb794

Do I need to run scaladoc before running the generator (as part of compile)?

**Comment by andyglow** [comment]
well. the problem is that
there might be several sources of schema, right?
1. the source code of the project you are working on
2. dependencies

If i get you right, you have described the second situation.
In this case the solution might be to pack docs alongside with bytecode, by placing them into a java annotation for example. this can be done by scala compiler plugin. but again - you have to be able to recompile that dependencies, i mean you have to own them.

i am working part time on this project here
https://github.com/andyglow/scaladocx
it aims to provide exactly what i've said.

till then only models maintained as part of the project supported, not the dependent ones.

**Comment by gordol** [comment]
@andyglow

does this currently work? I see related commits, but it seems to be non-functional in my project. the scaladocs work fine... i can generate docs and it generates HTML as expected, but the descriptions aren't getting pulled into the schemas by scala-jsonschema.

does this not work for imports? even if they are in the same namespace as the package that i'm building, in my own repo, alongside the main, but in a different file?

when you say that this only works for the models as part of the project that we maintain, does it not work if our project uses multiple classes and package namespaces?

if i make a simple example, as shown below, where the comments are in the same file as the one where the JsonFormatter is ran, then it appears to work, but if FooBar was imported from `com.gordol.foobar.*`, it fails to work.

**DISREGARD THE ABOVE** 

i started to build a new project as a POC to explain the weird behavior, with some different namespaces and submodules, and... instead of sharing it here, i'll just explain, because it's obvious in hindsight.

okay, so i think i figured it out... i kept having very strange behavior, and it would work sometimes, and not others. it only worked when i made a new module, for the first time...

so on a hunch, i ran `sbt clean` and SUCCESS!!! it now generates all of the description fields based on the docstrings.

so, anyone here, if you have problems with it not working, make sure you run `sbt clean` first.

given the discussion above, this makes total sense, since sbt is caching the built module bytecode i'm sure... and this library only picks up the comments the first time it's compiled.

hope this helps someone!

**Comment by andyglow** [comment]
There is an idea to incorporate a compiler plugin that would carry scaladocs as java annotations, so that downstream transformations had all that and information wouldn't be lost. But that project is on hold. I don't know when I will have time to get back to it.

**Comment by gordol** [comment]
hi @andyglow i appreciate the quick response.

in the meantime, can we decorate/annotate the fields in the classes to hold the descriptions?

a similar library for avro schemas supports this: https://github.com/sksamuel/avro4s#adding-properties-and-docs-to-a-schema.

**Comment by andyglow** [comment]
hi, guys..
please check https://github.com/andyglow/scala-jsonschema#documentation
this is going to be available in 0.4.0
